lib/ansible/modules/source_control/git_config.py

```diff
@@ -209,10 +234,20 @@ def main():
     if params['list_all'] and scope and rc == 128 and 'unable to read config file' in err:
         # This just means nothing has been set at the given scope
         module.exit_json(changed=False, msg='', config_values={})
-    elif rc >= 2:
+    elif rc >= 2 and not config_state:
         # If the return code is 1, it just means the option hasn't been set yet, which is fine.
         module.fail_json(rc=rc, msg=err, cmd=' '.join(args))
 
+    if config_state and out:
+        # if the setting is present, insert --unset in args list and .run_command
+        old_val = out.rstrip()
+        args.insert(-1, "--unset")
+        (rc, out, err) = module.run_command(' '.join(args), cwd=dir)
```

Comment:
This code does not account for check mode.

Reply:
Understood - I'll continue work on `module.check_mode` for this PR. Thanks again for the feedback.